core/capabilities/remote/target/caller.go

```diff
@@ -0,0 +1,142 @@
+package target
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/target/request"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// caller is a shim for remote target capabilities.
+// It translates between capability API calls and network messages.
+// Its responsibilities are:
+//  1. Transmit capability requests to remote nodes according to a transmission schedule
+//  2. Aggregate responses from remote nodes and return the aggregated response
+//
+// caller communicates with corresponding receiver on remote nodes.
+type caller struct {
```

Comment:
IMO caller isn't a great name :( It makes it unclear about whether this is the object responsible for being the shim (i.e. on the local node) or for being the "bridge" (the object that propagates network calls to the remote capability.

Reply:
Aside: it would be really great if we could have some clear naming conventions for these objects across all capability types

Eg. for triggers we have a subscriber (on the local node) and a publisher (on the remote); here we have a Caller (on the local node) and a receiver (on the remote)

What's slightly confusing about caller + receiver is that actually we communicate both ways (i.e. the receiver is also a caller when it's transmitting results back to the local node)

Reply:
don't disagree, lacked the imagination to come up with a better name that didn;t clash with other names already taken. with package name its  remote.target.caller.   Open to suggestions though!

Reply:
renamed to client/server - an overloaded term, but it fits better I think